pkgs/applications/misc/calcoo/default.nix

```diff
@@ -3,34 +3,33 @@
 , fetchzip
 , ant
 , jdk
+, jre
 , makeWrapper
+, canonicalize-jars-hook
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "calcoo";
   version = "2.1.0";
 
   src = fetchzip {
-    url = "mirror://sourceforge/project/calcoo/calcoo/${version}/${pname}-${version}.zip";
+    url = "mirror://sourceforge/calcoo/calcoo-${finalAttrs.version}.zip";
     hash = "sha256-Bdavj7RaI5CkWiOJY+TPRIRfNelfW5qdl/74J1KZPI0=";
   };
 
-  patches = [
-    # Sets javac encoding option on build.xml
-    ./0001-javac-encoding.diff
-  ];
-
   nativeBuildInputs = [
     ant
     jdk
     makeWrapper
+    canonicalize-jars-hook
   ];
 
   dontConfigure = true;
 
   buildPhase = ''
     runHook preBuild
 
+    export JAVA_TOOL_OPTIONS="-Dfile.encoding=iso-8859-1"
```

Comment:
It removes the need for the patch file. Nice idea!

However I prefer the more explicit `env.JAVA_TOOL_OPTIONS = "-Dfile.encoding=iso-8859-1";`. Can you verify if it works?

```suggestion
```

Reply:
It works fine